text/0116-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,369 @@
+```
+
+This is compiled to:
+
+```css
+div.x-foo_foo > .bar.x-foo_foo {}
+```
+
+Any existing classes in the template will be merged with the scoping token classes.
+
+### Comparisons with synthetic shadow DOM
+
+The current design of LWC's synthetic shadow scoped styles has several features that we do not want to emulate with light DOM scoped styles:
+
+- Using attributes over classes.
+- Allowing `@import` to import other CSS files.
+- Using `lwc:dom="manual"` and `MutationObserver` to dynamically update the scoping token for new DOM nodes.
+
+The classes vs attribute issue is already addressed above. So let's cover the other differences.
+
+#### `@import`
+
+For the time being, `@import` will be disallowed within `*.scoped.css` files. It may be enabled in the future, but for now, the invariant that "only files ending in `.scoped.css` are scoped" makes the implementation much simpler, and matches other scoped CSS implementations such as Vue and Svelte, which do not allow `@import`. (In addition, native constructable stylesheets [do not support `@import`](https://chromestatus.com/feature/4735925877735424).)
+
+#### `lwc:dom="manual"` and `MutationObserver`
+
+Consider this example:
+
+```html
+<!-- example.html -->
+<template>
+  <div lwc:dom="manual"></div>
+</template>
+```
+
+```css
+/* example.css */
+span {
+    color: green;
+}
+```
+
+```js
+// example.js
+class Example extends LightningElement {
+  renderedCallback() {
+    this.template.querySelector('div').appendChild(document.createElement('span'))
+  }
+}
+```
+
+In this case, the `<span>` is dynamically inserted into the component's shadow DOM, but synthetic shadow DOM needs some way to monitor the DOM for changes so that it can apply the styling token. This is where `lwc:dom="manual"` comes in – it's a signal to add a `MutationObserver` to track changes.
+
+This is a lot of extra machinery to support style scoping. Users need to know about `lwc:dom="manual"`, and the framework needs to create and disconnect a `MutationObserver`.
+
+We would like to avoid this for light DOM style scoping, so we have a simpler system: dynamically-inserted elements are not scoped. Incidentally, this is how Vue and Svelte scoped styles work – there's no expectation that you can mutate the DOM with vanilla DOM APIs and still have the scoping token applied.
+
+### Targeting the root element
+
+With global light DOM styles, the `:host` selector is inserted as-is. This means that a light DOM child component can use `:host` to target its shadow parent's host container.
+
+With scoped styles, the intuition around `:host` should be a bit different. Rather than inserting the CSS as-is, scoped styles imitate the encapsulation of shadow DOM components. So it makes sense for `:host` to follow shadow DOM-like semantics. This means that `:host` should refer to the root element of the light DOM component – i.e., the root of the scoped DOM region.
+
+So for instance:
+
+```css
+/* light.css */
+:host {

```

Comment:
@nolanlawson I find this a little counter-intuitive: 

"I'm a shadow friendly component, my styles are robust and encapsulated, I do allow others externally to apply some styles to the `host` element, as I understand I'm only a guest for a certain point and time in the cycle, but I would never expect a child component of my own to be able to mess with my styles"